lib/pbench/test/unit/server/test_relay.py

```diff
@@ -0,0 +1,318 @@
+                "metadata": [],
+            },
+        )
+        responses.add(
+            responses.GET, "https://relay.example.com/uri2", status=HTTPStatus.NOT_FOUND
+        )
+        response = client.post(
+            self.gen_uri(server_config, "https://relay.example.com/uri1"),
+            headers=self.gen_headers(pbench_drb_token),
+        )
+        assert (
+            response.status_code == HTTPStatus.NOT_FOUND
+        ), f"Unexpected result, {response.text}"
+
+        audit = Audit.query()
+        assert len(audit) == 2
+        assert audit[0].id == 1
+        assert audit[0].root_id is None
+        assert audit[0].operation == OperationCode.CREATE
+        assert audit[0].status == AuditStatus.BEGIN
+        assert audit[0].name == "upload"
+        assert audit[0].object_type == AuditType.DATASET
+        assert audit[0].object_id == md5
+        assert audit[0].object_name == Dataset.stem(file)
+        assert audit[0].user_id == DRB_USER_ID
+        assert audit[0].user_name == "drb"
+        assert audit[0].reason is None
+        assert audit[0].attributes == {
+            "access": "private",
+            "metadata": {},
+        }
+        assert audit[1].id == 2
+        assert audit[1].root_id == 1
+        assert audit[1].operation == OperationCode.CREATE
+        assert audit[1].status == AuditStatus.FAILURE
+        assert audit[1].name == "upload"
+        assert audit[1].object_type == AuditType.DATASET
+        assert audit[1].object_id == md5
+        assert audit[1].object_name == Dataset.stem(file)
+        assert audit[1].user_id == DRB_USER_ID
+        assert audit[1].user_name == "drb"
+        assert audit[1].reason == AuditReason.CONSISTENCY
+        assert audit[1].attributes == {
+            "message": "Unable to retrieve relay tarball: 'Not Found'"
+        }
+
+    @responses.activate
+    def test_relay_no_json(self, client, server_config, pbench_drb_token, tarball):
+        file, md5file, md5 = tarball
+        responses.add(
+            responses.GET, "https://relay.example.com/uri1", status=HTTPStatus.NOT_FOUND
+        )
+        responses.add(
+            responses.GET, "https://relay.example.com/uri2", status=HTTPStatus.NOT_FOUND
+        )
+        response = client.post(
+            self.gen_uri(server_config, "https://relay.example.com/uri1"),
+            headers=self.gen_headers(pbench_drb_token),
+        )
+        assert (
+            response.status_code == HTTPStatus.BAD_GATEWAY
+        ), f"Unexpected result, {response.text}"
+
+    @responses.activate
+    def test_relay_not_json(self, client, server_config, pbench_drb_token):
+        responses.add(
+            responses.GET,
+            "https://relay.example.com/uri1",
+            status=HTTPStatus.OK,
+            body="This isn't JSON",
+        )
+        responses.add(
+            responses.GET, "https://relay.example.com/uri2", status=HTTPStatus.NOT_FOUND
+        )
```

Comment:
The response at line 236 should never be used, right?

So, rather than setting it to `HTTPStatus.NOT_FOUND`, we should be rigging it to cause the test to fail.  (E.g., if you _don't_ provide the response, then I think it defaults to raising `ConnectionError`; alternately you can have the response be an [exception of your choosing](https://github.com/getsentry/responses/blob/master/README.rst#id12), such as `AssertionError`.)

Ditto line 263, et al.